### PR TITLE
chore(ci): add randomized long run tests on CPU and GPU

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
CPU & GPU tests long run tests are passing

File: tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_random_op_sequence.rs [hunk 1]
```diff
@@ -0,0 +1,1195 @@
...
+    let clear_left_shift = |x, y| x << y;
+    // Warning this rotate definition only works with 64-bit ciphertexts
+    let clear_rotate_right = |x: u64, y: u64| x.rotate_right(y as u32);
+    let clear_right_shift = |x, y| x >> y;
+    let clear_max = |x: u64, y: u64| max(x, y);
+    let clear_min = |x: u64, y: u64| min(x, y);
+
+    #[allow(clippy::type_complexity)]
+    let mut binary_ops: Vec<(BinaryOpExecutor, &dyn Fn(u64, u64) -> u64, String)> = vec![
+        (Box::new(add_executor), &clear_add, "add".parse().unwrap()),
```
Comment (tmontaigu) [inline tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_random_op_sequence.rs:90]: Should replace all `"some_str.parse().unwrap()"` with `"some_str".to_string()`

File: tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_random_op_sequence.rs [hunk 2]
```diff
@@ -0,0 +1,1195 @@
...
+        &mut scalar_comparison_ops,
+        &mut select_op,
+        &mut div_rem_op,
+        &mut scalar_div_rem_op,
+        &mut log2_ops,
+    );
+}
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn random_op_sequence_test<P>(
```
Comment (tmontaigu) [inline tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_random_op_sequence.rs:441]: One suggestion to simplify a bit checks in the if and the subtraction done to get the index:

```rust
    let binary_ops_range = 0..binary_ops.len();
    let unary_ops_range = binary_ops_range.end..binary_ops_range.len()+unary_ops.len();
    // ..etc

    let index = 0;
    if binary_ops_range.contains(&index) {
        let i = index - binary_ops_range.start;
    } else if unary_ops_range.contains(&index) {
        let i = index - unary_ops_range.start;
    } // etc...
```

or, pick the thin using 2 randoms: one to get the op type, then another to get the op


```rust
let op_type_index = gen.range(0..num_op_type);
match op_type_index {
0 => { 
    let op_index = en.range(0..binary_ops.len());
   /...
}
1 => { 
    let op_index = en.range(0..unary_ops.len());
   /...
}
// ..
_ => panic!("invalid op type index")
```

File: tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_signed_erc20.rs [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+    );
+}
+
+pub(crate) fn signed_safe_erc20_test<P, T1, T2, T3, T4>(
+    param: P,
+    mut overflowing_add_executor: T1,
+    mut overflowing_sub_executor: T2,
+    mut if_then_else_executor: T3,
+    mut bitor_executor: T4,
+) where
```
[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_signed_erc20.rs:102]: Should I just remove those two tests then?

[the remark this answers is not in the document]
Comment (agnesLeroy) [reply tfhe/src/integer/server_key/radix_parallel/tests_long_run/test_signed_erc20.rs:102]: I removed them

**Review by tmontaigu — APPROVED** [review]

**Pull request merged**